modules/controlpanel.cpp

```diff
@@ -619,6 +619,35 @@ class CAdminMod : public CModule {
         }
     }
 
+    void CloneNetwork(const CString& sLine) {
+        const CString sUsername = sLine.Token(1);
+        const CString sOldNetwork = sLine.Token(2);
+        const CString sNewNetwork = sLine.Token(3);
+
+        if (sOldNetwork.empty() || sNewNetwork.empty()) {
+            PutModule(
+                "Usage: CloneNetwork <username> <old network name> <new network name>");
+            return;
+        }
+
+        CUser* pUser = FindUser(sUsername);
+        if (!pUser) {
+            PutModule("Error: User [" + sUsername + "] not found.");
+            return;
+        }
+
+        CIRCNetwork* pOldNetwork = FindNetwork(pUser, sOldNetwork);
+        if (!pOldNetwork) {
+            PutModule("Error: Network [" + sOldNetwork + "] not found.");
+            return;
+        }
+
+        CIRCNetwork* pNewNetwork = new CIRCNetwork(pUser, *pOldNetwork);
+        pNewNetwork->SetName(sNewNetwork);
```

Comment:
> Is this what you meant?

`SetName` can fail. Please look at the following and handle this function's return value:
https://github.com/znc/znc/blob/f885699d1a3cb6376dc6162f9e8440791034a329/src/IRCNetwork.cpp#L771-L778